src/sdk/namespace_fs.js

```diff
@@ -79,6 +79,46 @@ function sort_entries_by_name(a, b) {
     return 0;
 }
 
+function _get_mtime_from_filename(filename) {
+    if (!filename.includes('_mtime-')) {
+        // Latest file wont have time suffix which will push the latest
+        // object last in the list. So to keep the order maintained,
+        // returning the latest time. Multiplying with 1e6 to provide
+        // nano second precision
+        return Date.now() * 1e6;
+    }
+    const file_parts = filename.split('-');
+    return parseInt(file_parts[1], 36);
+}
+
+function _get_filename(file_name) {
+    if (file_name.includes('_mtime-')) {
+        return file_name.substring(0, file_name.indexOf('_mtime-'));
+    }
+    return file_name;
+}
+/**
+ * @param {fs.Dirent} a
+ * @param {fs.Dirent} b
+ * @returns {Number}
+ */
+function sort_entries_by_name_and_time(a, b) {
+    // const a_mtime = _get_mtime_from_filename(a.name);
+    // const b_mtime = _get_mtime_from_filename(b.name);
+    // return b_mtime - a_mtime;
+    const a_name = _get_filename(a.name);
+    const b_name = _get_filename(b.name);
+    if (a_name === b_name) {
+        const a_mtime = _get_mtime_from_filename(a.name);
+        const b_mtime = _get_mtime_from_filename(b.name);
+        return b_mtime - a_mtime;
```

Comment:
I'm not sure native subtraction will work with BigInt. you can use `copmare`
```suggestion
        return b_mtime.compare(a_mtime);
```

Reply:
This is also not supported with BigInt()